### ISSUE TITLE
`isolated-functions`: Should allow globals configured with `/* global */` comments

### ISSUE THREAD

**Issue by andersk** [sindresorhus/eslint-plugin-unicorn#3492]
### Description

`isolated-functions` should allow global variables according to the [documentation](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/isolated-functions.md), but it fails to allow global variables configured with a [`/* global */` comment](https://eslint.org/docs/latest/use/configure/language-options#use-configuration-comments).

### Examples

```js
/* global foo */

/** @isolated */
export const f = () => foo; // should be allowed
```
